// @novu/node/build/module/lib/layouts/layouts.js
import { WithHttp } from '../novu.interface';
export class Layouts extends WithHttp {
    async list(data) {
        return await this.http.get(`/layouts`, {
            params: {
                page: data.page,
                ...(data?.pageSize && { pageSize: data.pageSize }),
                ...(data?.sortBy && { sortBy: data.sortBy }),
                ...(data?.orderBy && { orderBy: data.orderBy }),
            },
        });
    }
    async get(layoutId) {
        return await this.http.get(`/layouts`, {
            params: {
                layoutId,
            },
        });
    }
    async create(data) {
        return await this.http.post(`/layouts`, {
            name: data.name,
            description: data.description,
            content: data.content,
            variables: data.variables,
            isDefault: data.isDefault,
        });
    }
    async update(layoutId, data) {
        return await this.http.patch(`/layouts/${layoutId}`, {
            ...(data.name && { name: data.name }),
            ...(data.description && { description: data.description }),
            ...(data.content && { content: data.content }),
            ...(data.variables && { variables: data.variables }),
            ...(typeof data.isDefault === 'boolean' && { isDefault: data.isDefault }),
        });
    }
    async delete(layoutId) {
        return await this.http.delete(`/layouts/${layoutId}`);
    }
    async setDefault(layoutId) {
        return await this.http.post(`/layouts/${layoutId}/default`);
    }
}
