// @novu/shared/dist/cjs/consts/providers/channels/chat.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.chatProviders = void 0;
const credentials_1 = require("../credentials");
const provider_enum_1 = require("../provider.enum");
const types_1 = require("../../../types");
exports.chatProviders = [
    {
        id: provider_enum_1.ChatProviderIdEnum.Slack,
        displayName: 'Slack',
        channel: types_1.ChannelTypeEnum.CHAT,
        credentials: credentials_1.slackConfig,
        docReference: 'https://docs.novu.co/channels/chat/slack',
        logoFileName: { light: 'slack.svg', dark: 'slack.svg' },
    },
    {
        id: provider_enum_1.ChatProviderIdEnum.Discord,
        displayName: 'Discord',
        channel: types_1.ChannelTypeEnum.CHAT,
        credentials: [],
        docReference: 'https://docs.novu.co/channels/chat/discord',
        logoFileName: { light: 'discord.svg', dark: 'discord.svg' },
    },
    {
        id: provider_enum_1.ChatProviderIdEnum.MsTeams,
        displayName: 'MSTeams',
        channel: types_1.ChannelTypeEnum.CHAT,
        credentials: [],
        docReference: 'https://docs.novu.co/channels/chat/msteams',
        logoFileName: { light: 'msteams.svg', dark: 'msteams.svg' },
    },
    {
        id: provider_enum_1.ChatProviderIdEnum.Mattermost,
        displayName: 'Mattermost',
        channel: types_1.ChannelTypeEnum.CHAT,
        credentials: [],
        docReference: 'https://developers.mattermost.com/integrate/webhooks/incoming/',
        logoFileName: { light: 'mattermost.svg', dark: 'mattermost.svg' },
    },
];
