// @novu/shared/dist/cjs/consts/providers/channels/email.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.emailProviders = void 0;
const credentials_1 = require("../credentials");
const provider_enum_1 = require("../provider.enum");
const types_1 = require("../../../types");
exports.emailProviders = [
    {
        id: provider_enum_1.EmailProviderIdEnum.Mailgun,
        displayName: 'Mailgun',
        channel: types_1.ChannelTypeEnum.EMAIL,
        credentials: credentials_1.mailgunConfig,
        docReference: 'https://docs.novu.co/channels/email/mailgun',
        logoFileName: { light: 'mailgun.svg', dark: 'mailgun.svg' },
    },
    {
        id: provider_enum_1.EmailProviderIdEnum.Mailjet,
        displayName: 'Mailjet',
        channel: types_1.ChannelTypeEnum.EMAIL,
        credentials: credentials_1.mailjetConfig,
        docReference: 'https://docs.novu.co/channels/email/mailjet',
        logoFileName: { light: 'mailjet.png', dark: 'mailjet.png' },
    },
    {
        id: provider_enum_1.EmailProviderIdEnum.Mandrill,
        displayName: 'Mandrill',
        channel: types_1.ChannelTypeEnum.EMAIL,
        credentials: credentials_1.mandrillConfig,
        docReference: 'https://docs.novu.co/channels/email/mandrill',
        logoFileName: { light: 'mandrill.svg', dark: 'mandrill.svg' },
    },
    {
        id: provider_enum_1.EmailProviderIdEnum.Postmark,
        displayName: 'Postmark',
        channel: types_1.ChannelTypeEnum.EMAIL,
        credentials: credentials_1.postmarkConfig,
        docReference: 'https://docs.novu.co/channels/email/postmark',
        logoFileName: { light: 'postmark.png', dark: 'postmark.png' },
    },
    {
        id: provider_enum_1.EmailProviderIdEnum.SendGrid,
        displayName: 'SendGrid',
        channel: types_1.ChannelTypeEnum.EMAIL,
        credentials: credentials_1.sendgridConfig,
        docReference: 'https://docs.novu.co/channels/email/sendgrid',
        logoFileName: { light: 'sendgrid.png', dark: 'sendgrid.png' },
    },
    {
        id: provider_enum_1.EmailProviderIdEnum.Sendinblue,
        displayName: 'Sendinblue',
        channel: types_1.ChannelTypeEnum.EMAIL,
        credentials: credentials_1.sendinblueConfig,
        docReference: 'https://docs.novu.co/channels/email/sendinblue',
        logoFileName: { light: 'sendinblue.png', dark: 'sendinblue.png' },
    },
    {
        id: provider_enum_1.EmailProviderIdEnum.SES,
        displayName: 'SES',
        channel: types_1.ChannelTypeEnum.EMAIL,
        credentials: credentials_1.sesConfig,
        docReference: 'https://docs.novu.co/channels/email/ses',
        logoFileName: { light: 'ses.svg', dark: 'ses.svg' },
    },
    {
        id: provider_enum_1.EmailProviderIdEnum.NetCore,
        displayName: 'NetCore',
        channel: types_1.ChannelTypeEnum.EMAIL,
        credentials: credentials_1.netCoreConfig,
        docReference: 'https://docs.novu.co/channels/email/netcore',
        logoFileName: { light: 'netcore.png', dark: 'netcore.png' },
    },
    {
        id: provider_enum_1.EmailProviderIdEnum.CustomSMTP,
        displayName: 'Custom SMTP',
        channel: types_1.ChannelTypeEnum.EMAIL,
        credentials: credentials_1.nodemailerConfig,
        docReference: 'https://docs.novu.co/channels/email/custom-smtp',
        logoFileName: { light: 'custom_smtp.svg', dark: 'custom_smtp.svg' },
    },
    {
        id: provider_enum_1.EmailProviderIdEnum.MailerSend,
        displayName: 'MailerSend',
        channel: types_1.ChannelTypeEnum.EMAIL,
        credentials: credentials_1.mailerSendConfig,
        docReference: 'https://docs.novu.co/channels/email/mailersend',
        logoFileName: { light: 'mailersend.svg', dark: 'mailersend.svg' },
    },
    {
        id: provider_enum_1.EmailProviderIdEnum.Outlook365,
        displayName: 'Microsoft Outlook365',
        channel: types_1.ChannelTypeEnum.EMAIL,
        credentials: credentials_1.outlook365Config,
        docReference: 'https://docs.novu.co/channels/email/outlook365',
        logoFileName: { light: 'outlook365.png', dark: 'outlook365.png' },
    },
    {
        id: provider_enum_1.EmailProviderIdEnum.Infobip,
        displayName: 'Infobip',
        channel: types_1.ChannelTypeEnum.EMAIL,
        credentials: credentials_1.infobipEmailConfig,
        docReference: 'https://docs.novu.co/channels/email/infobip',
        logoFileName: { light: 'infobip.png', dark: 'infobip.png' },
    },
    {
        id: provider_enum_1.EmailProviderIdEnum.Resend,
        displayName: 'Resend',
        channel: types_1.ChannelTypeEnum.EMAIL,
        credentials: credentials_1.resendConfig,
        docReference: 'https://docs.novu.co/channels/email/resend',
        logoFileName: { light: 'resend.svg', dark: 'resend.svg' },
    },
    {
        id: provider_enum_1.EmailProviderIdEnum.SparkPost,
        displayName: 'SparkPost',
        channel: types_1.ChannelTypeEnum.EMAIL,
        credentials: credentials_1.sparkpostConfig,
        docReference: 'https://docs.novu.co/channels/email/sparkpost',
        logoFileName: { light: 'sparkpost.svg', dark: 'sparkpost.svg' },
    },
    {
        id: provider_enum_1.EmailProviderIdEnum.EmailWebhook,
        displayName: 'Email Webhook',
        channel: types_1.ChannelTypeEnum.EMAIL,
        credentials: credentials_1.emailWebhookConfig,
        betaVersion: true,
        docReference: 'https://docs.novu.co/channels/email/email-webhook/',
        logoFileName: { light: 'email_webhook.svg', dark: 'email_webhook.svg' },
    },
];
