// @novu/shared/dist/cjs/consts/providers/channels/sms.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.smsProviders = void 0;
const credentials_1 = require("../credentials");
const provider_enum_1 = require("../provider.enum");
const types_1 = require("../../../types");
exports.smsProviders = [
    {
        id: provider_enum_1.SmsProviderIdEnum.Nexmo,
        displayName: 'Nexmo',
        channel: types_1.ChannelTypeEnum.SMS,
        credentials: credentials_1.nexmoConfig,
        docReference: 'https://docs.novu.co/channels/sms/nexmo',
        logoFileName: { light: 'nexmo.png', dark: 'nexmo.png' },
    },
    {
        id: provider_enum_1.SmsProviderIdEnum.Plivo,
        displayName: 'Plivo',
        channel: types_1.ChannelTypeEnum.SMS,
        credentials: credentials_1.plivoConfig,
        docReference: 'https://www.plivo.com/docs/',
        logoFileName: { light: 'plivo.png', dark: 'plivo.png' },
    },
    {
        id: provider_enum_1.SmsProviderIdEnum.Sms77,
        displayName: 'sms77',
        channel: types_1.ChannelTypeEnum.SMS,
        credentials: credentials_1.sms77Config,
        docReference: 'https://docs.novu.co/channels/sms/SMS77',
        logoFileName: { light: 'sms77.svg', dark: 'sms77.svg' },
    },
    {
        id: provider_enum_1.SmsProviderIdEnum.SNS,
        displayName: 'SNS',
        channel: types_1.ChannelTypeEnum.SMS,
        credentials: credentials_1.snsConfig,
        docReference: 'https://docs.novu.co/channels/sms/sns',
        logoFileName: { light: 'sns.svg', dark: 'sns.svg' },
    },
    {
        id: provider_enum_1.SmsProviderIdEnum.Telnyx,
        displayName: 'Telnyx',
        channel: types_1.ChannelTypeEnum.SMS,
        credentials: credentials_1.telnyxConfig,
        docReference: 'https://docs.novu.co/channels/sms/telnyx',
        logoFileName: { light: 'telnyx.png', dark: 'telnyx.png' },
    },
    {
        id: provider_enum_1.SmsProviderIdEnum.Twilio,
        displayName: 'Twilio',
        channel: types_1.ChannelTypeEnum.SMS,
        credentials: credentials_1.twilioConfig,
        docReference: 'https://docs.novu.co/channels/sms/twilio',
        logoFileName: { light: 'twilio.png', dark: 'twilio.png' },
    },
    {
        id: provider_enum_1.SmsProviderIdEnum.Gupshup,
        displayName: 'Gupshup',
        channel: types_1.ChannelTypeEnum.SMS,
        credentials: credentials_1.gupshupConfig,
        docReference: 'https://docs.gupshup.io/docs/send-single-message',
        logoFileName: { light: 'gupshup.png', dark: 'gupshup.png' },
    },
    {
        id: provider_enum_1.SmsProviderIdEnum.Firetext,
        displayName: 'Firetext',
        channel: types_1.ChannelTypeEnum.SMS,
        credentials: credentials_1.firetextConfig,
        docReference: 'https://www.firetext.co.uk/docs',
        logoFileName: { light: 'firetext.svg', dark: 'firetext.svg' },
    },
    {
        id: provider_enum_1.SmsProviderIdEnum.Infobip,
        displayName: 'Infobip',
        channel: types_1.ChannelTypeEnum.SMS,
        credentials: credentials_1.infobipSMSConfig,
        docReference: 'https://www.infobip.com/docs',
        logoFileName: { light: 'infobip.png', dark: 'infobip.png' },
    },
    {
        id: provider_enum_1.SmsProviderIdEnum.BurstSms,
        displayName: 'BurstSMS',
        channel: types_1.ChannelTypeEnum.SMS,
        credentials: credentials_1.burstSmsConfig,
        docReference: 'https://developer.transmitsms.com/',
        logoFileName: { light: 'burst-sms.svg', dark: 'burst-sms.svg' },
    },
    {
        id: provider_enum_1.SmsProviderIdEnum.Clickatell,
        displayName: 'clickatell',
        channel: types_1.ChannelTypeEnum.SMS,
        credentials: credentials_1.clickatellConfig,
        betaVersion: true,
        docReference: 'https://docs.clickatell.com/',
        logoFileName: { light: 'clickatell.png', dark: 'clickatell.png' },
    },
    {
        id: provider_enum_1.SmsProviderIdEnum.FortySixElks,
        displayName: '46elks',
        channel: types_1.ChannelTypeEnum.SMS,
        credentials: credentials_1.fortySixElksConfig,
        docReference: 'https://46elks.com/docs/send-sms',
        logoFileName: { light: '46elks.png', dark: '46elks.png' },
    },
    {
        id: provider_enum_1.SmsProviderIdEnum.Kannel,
        displayName: 'Kannel SMS',
        channel: types_1.ChannelTypeEnum.SMS,
        credentials: credentials_1.kannelConfig,
        betaVersion: true,
        docReference: 'https://www.kannel.org/doc.shtml',
        logoFileName: { light: 'kannel.png', dark: 'kannel.png' },
    },
    {
        id: provider_enum_1.SmsProviderIdEnum.Maqsam,
        displayName: 'Maqsam',
        channel: types_1.ChannelTypeEnum.SMS,
        credentials: credentials_1.maqsamConfig,
        docReference: 'https://portal.maqsam.com/docs/v2/sms',
        logoFileName: { light: 'maqsam.png', dark: 'maqsam.png' },
    },
    {
        id: provider_enum_1.SmsProviderIdEnum.SmsCentral,
        displayName: 'SMS Central',
        channel: types_1.ChannelTypeEnum.SMS,
        credentials: credentials_1.smsCentralConfig,
        docReference: 'https://www.smscentral.com.au/sms-api/',
        logoFileName: { light: 'sms-central.png', dark: 'sms-central.png' },
    },
    {
        id: provider_enum_1.SmsProviderIdEnum.Termii,
        displayName: 'Termii',
        channel: types_1.ChannelTypeEnum.SMS,
        credentials: credentials_1.termiiConfig,
        docReference: 'https://developers.termii.com/authentication',
        logoFileName: { light: 'termii.png', dark: 'termii.png' },
    },
    {
        id: provider_enum_1.SmsProviderIdEnum.AfricasTalking,
        displayName: `Africa's Talking`,
        channel: types_1.ChannelTypeEnum.SMS,
        credentials: credentials_1.africasTalkingConfig,
        docReference: 'https://docs.novu.co/channels/sms/africas-talking',
        logoFileName: { light: 'africas-talking.svg', dark: 'africas-talking.svg' },
    },
];
