// @novu/node/build/main/lib/layouts/layouts.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (g && (g = 0, op[0] && (_ = 0)), _) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Layouts = void 0;
var novu_interface_1 = require("../novu.interface");
var Layouts = (function (_super) {
    __extends(Layouts, _super);
    function Layouts() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Layouts.prototype.list = function (data) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4, this.http.get("/layouts", {
                            params: __assign(__assign(__assign({ page: data.page }, ((data === null || data === void 0 ? void 0 : data.pageSize) && { pageSize: data.pageSize })), ((data === null || data === void 0 ? void 0 : data.sortBy) && { sortBy: data.sortBy })), ((data === null || data === void 0 ? void 0 : data.orderBy) && { orderBy: data.orderBy })),
                        })];
                    case 1: return [2, _a.sent()];
                }
            });
        });
    };
    Layouts.prototype.get = function (layoutId) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4, this.http.get("/layouts", {
                            params: {
                                layoutId: layoutId,
                            },
                        })];
                    case 1: return [2, _a.sent()];
                }
            });
        });
    };
    Layouts.prototype.create = function (data) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4, this.http.post("/layouts", {
                            name: data.name,
                            description: data.description,
                            content: data.content,
                            variables: data.variables,
                            isDefault: data.isDefault,
                        })];
                    case 1: return [2, _a.sent()];
                }
            });
        });
    };
    Layouts.prototype.update = function (layoutId, data) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4, this.http.patch("/layouts/".concat(layoutId), __assign(__assign(__assign(__assign(__assign({}, (data.name && { name: data.name })), (data.description && { description: data.description })), (data.content && { content: data.content })), (data.variables && { variables: data.variables })), (typeof data.isDefault === 'boolean' && { isDefault: data.isDefault })))];
                    case 1: return [2, _a.sent()];
                }
            });
        });
    };
    Layouts.prototype.delete = function (layoutId) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4, this.http.delete("/layouts/".concat(layoutId))];
                    case 1: return [2, _a.sent()];
                }
            });
        });
    };
    Layouts.prototype.setDefault = function (layoutId) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4, this.http.post("/layouts/".concat(layoutId, "/default"))];
                    case 1: return [2, _a.sent()];
                }
            });
        });
    };
    return Layouts;
}(novu_interface_1.WithHttp));
exports.Layouts = Layouts;
