// @novu/node/build/module/lib/notification-template/notification-template.js
import { WithHttp } from '../novu.interface';
export class NotificationTemplates extends WithHttp {
    async getAll(page, limit) {
        if (page === undefined && limit === undefined) {
            return await this.http.get(`/notification-templates`, {});
        }
        else if (page === undefined) {
            return await this.http.get(`/notification-templates`, {
                params: {
                    limit,
                },
            });
        }
        else if (limit === undefined) {
            return await this.http.get(`/notification-templates`, {
                params: { page },
            });
        }
        else {
            return await this.http.get(`/notification-templates`, {
                params: { page, limit },
            });
        }
    }
    async create(data) {
        return await this.http.post(`/notification-templates`, {
            ...data,
        });
    }
    async update(templateId, data) {
        return await this.http.put(`/notification-templates/${templateId}`, {
            ...data,
        });
    }
    async delete(templateId) {
        return await this.http.delete(`/notification-templates/${templateId}`);
    }
    async getOne(templateId) {
        return await this.http.get(`/notification-templates/${templateId}`);
    }
    async updateStatus(templateId, active) {
        return await this.http.put(`/notification-templates/${templateId}/status`, {
            active,
        });
    }
}
